proxy/src/banner.rs

```diff
@@ -10,4 +12,14 @@ pub fn print_banner() {
         }
     };
     info!("🌈 Samply.Beam ({}) v{} (built {} {}, {}) starting up ...", env!("CARGO_PKG_NAME"), env!("CARGO_PKG_VERSION"), env!("BUILD_DATE"), env!("BUILD_TIME"), commit);
+}
+
+pub(crate) async fn set_server_header<B>(mut response: Response<B>) -> Response<B> {
+    let headers = response.headers_mut();
+    let target_header = match headers.contains_key(header::SERVER) {
+        true => header::VIA, // We're relaying for a proxys
```

Comment:
If I understand correctly, the VIA header should also be inserted for messages to the Broker, i.e. messages including a `header::USER_AGENT`. Additionally, the comment should read "We're relaying for a broker", no?

Reply:
Fixed in https://github.com/samply/beam/pull/87/commits/6a593967b8a30b099804d7ad457db17c3da6538b and https://github.com/samply/beam/pull/87/commits/418718996fdd2163989daef9a9620048607c3491